share/dkg/dkg.go

```diff
@@ -555,19 +559,25 @@ func (d *DistKeyGenerator) ExpectedResponsesFastSync() int {
 // - the justification bundle if this node must produce at least one. If nil,
 // this node must still wait on the justification phase.
 // - error if the dkg must stop now, an unrecoverable failure.
-func (d *DistKeyGenerator) ProcessResponses(bundles []*ResponseBundle) (*Result, *JustificationBundle, error) {
+func (d *DistKeyGenerator) ProcessResponses(bundles []*ResponseBundle) (res *Result, jb *JustificationBundle, err error) {
 	if !d.canReceive && d.state != DealPhase {
 		// if we are a old node that will leave
 		return nil, nil, fmt.Errorf("leaving node can process responses only after creating shares")
 	} else if d.state != ResponsePhase {
 		return nil, nil, fmt.Errorf("can only process responses after processing shares - current state %s", d.state)
 	}
 
+	defer func() {
```

Comment:
is there a benefit to deferring this vs just returning `return nil, bundle, d.checkIfEvicted(ResponsePhase)` at the bottom?

Reply:
ah, because it catches the naked `return` on 580 or?

Reply:
There are a lot of places where we return in the ProcessResponse method, so I would need to add it to all of them. That's why I figured named return here is actually useful to add a "wrapper" around the return value, if error is nil, I check if I'm evicted.